index.js

```diff
@@ -68,7 +68,31 @@ module.exports = function (str) {
   } else if ((match = str.match(new RegExp(regexps['re1'], 'i')))) {
     let days = match[1];
     let period = match[2];
+    if (period === 'year') {
+      let year = new Date().getFullYear() - days;
+      if (year < 0) {
+        return 'Before B.C.';
+      } else {
+        let updatedVal = new Date();
+        updatedVal.setFullYear(year);
+        return [true, updatedVal];
+      }
+    } else if (period === 'month') {
+      let extraYear = Math.floor(days / 12);
+      let year = new Date().getFullYear() - extraYear;
+      var month = new Date().getMonth();
+      let extraMonth = (month - days) % 12;
+      if (extraMonth >= 0) {
```

Comment:
The reminder of a division is never negative. This particular line isn't required. `extraMonth` is always `>= 0`

Reply:
But for the negative value of `extraMonth`, the `year` will also get reduced by one which is not true when `extraMonth` is positive.

Reply:
After line 84, `extraMonth` is positive. Your if/else is not required.

Reply:
((11-19)%12) = -8?

Reply:
`month` will be between 0 to 11 but `days` can be anything

Reply:
sir, may I get some response?

Reply:
Please stop spamming maintainers. One message is enough.

Reply:
It looks like I was mistaken. Sorry about that.

What I can see is that, `x % y >= 0 if x > 0`

but `-1 * y < x % y <= 0` if `x < 0`.